### PR TITLE
Batches refactor

### PR DESCRIPTION
**Pull request by mikemadden42**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -18,6 +19,20 @@ setup:
 	vagrant up
 	vagrant ssh-config > ssh_config
 
+
```
Comment (andrewkroh) [inline Makefile]: Extra new line.

File: Makefile [hunk 2]
```diff
@@ -18,6 +19,20 @@ setup:
...
+batch:
+	$(foreach GROUP,$(GROUPS),GROUP=$(GROUP) $(MAKE) run-group || exit 1;)
```
Comment (andrewkroh) [inline Makefile]: We have a mix of `${}` and `$()` styles. Let's converge on `${}`.

File: Makefile [hunk 3]
```diff
@@ -18,6 +19,20 @@ setup:
 	vagrant up
 	vagrant ssh-config > ssh_config
 
+
+batch:
```
Comment (andrewkroh) [inline Makefile]: Can you add comments to the new targets to describe what they do. And maybe add an example to the README like how to test only the windows group.

File: Makefile [hunk 4]
```diff
@@ -18,6 +19,20 @@ setup:
...
+
+run-group: HOSTS=$(shell ansible ${GROUP} -i hosts --list-hosts | tail -n +2)
+run-group:
+	vagrant up ${HOSTS}
+	vagrant status ${HOSTS}
+	vagrant ssh-config ${HOSTS} >ssh_config
+	ANSIBLE_LIMIT=${GROUP} make run
+	vagrant destroy -f ${HOSTS}
+
+.PHONY: batch clean run run-elastic run-group run-oss setup ve
```
Comment (andrewkroh) [inline Makefile]: Can you move this down to the bottom of the file. Feels like it should either be at the top or bottom of the file. Or break it down into one line `.PHONY: <target>` the preceeds each target.

```
.PHONY: run-group
run-group:
````

File: Makefile [hunk 5]
```diff
@@ -5,6 +5,7 @@ ANSIBLE_VERBOSE?=
 ANSIBLE_LIMIT?=all:!localhost
 # Extra flags to pass to Ansible (e.g. --skip-tags filebeat).
 ANSIBLE_EXTRA_FLAGS?=
+GROUPS?=centos debian sles windows
```
Comment (andrewkroh) [inline Makefile]: Can you add a description like these are the Ansible host groups declared in the `hosts` file.

File: Makefile (hunk 1 above)
Comment (mikemadden42) [reply Makefile]: New line has been removed.

File: Makefile (hunk 2 above)
Comment (mikemadden42) [reply Makefile]: I named all variables with ${VAR}.

File: Makefile (hunk 4 above)
Comment (mikemadden42) [reply Makefile]: I moved the phony targets to the end of the file.

**Pull request merged**